nautobot/extras/datasources/git.py

```diff
@@ -288,6 +291,14 @@ def ensure_git_repository(
     from_url, to_path, from_branch = get_repo_from_url_to_path_and_from_branch(
         repository_record, logger=logger, job_result=job_result
     )
+    # We want to check if the repo is already checked out at head. We also want to avoid calling
+    # nautobot.utilities.git.GitRepo.checkout, because it will cause the URL to be rebuilt causing calls to a secrets
+    # backend. As such, if head is None, we can't perform these checks.
```

Comment:
`get_repo_from_url_to_path_and_from_branch` is what is calling `repository_record.secrets_group.get_secret_value()` and constructing the url so I don't know if your comment is correct?

Reply:
Applied the fix from below, resolving this.

Reply:
Oh and I also changed the comment to properly reflect what is doing what.